### PR TITLE
+ Add UI and API logic to allow early bird presale by tx

### PR DESCRIPTION
**Pull request by williamchong007**
+ `isPreSale` flag in tokensale page
+ presale api and emails

https://likecoin-fdn.slack.com/messages/C9GQJ8X7C/files/F9RDEJ90Q/

### CODE REVIEW

File: pages/in/tokensale/index.vue [hunk 1]
```diff
@@ -57,6 +57,10 @@
               </div>
             </div>
           </div>
+          <div v-else class="tokensale-progress-wrapper lc-container-3 lc-verticle-inset-4">
```
Comment (nwingt) [inline pages/in/tokensale/index.vue]: Please create another class instead of using `tokensale-progress-wrapper`

**Comment by williamchong007** [comment]
@nwingt updated

File: pages/in/tokensale/index.vue [hunk 2]
```diff
@@ -544,14 +574,18 @@ export default {
...
 
-.tokensale-progress-wrapper {
+.tokensale-presale-wrapper, .tokensale-progress-wrapper {
   padding-right: 0;
   padding-left: 0;
 
   > div {
     margin-right: -8px;
     margin-left: -8px;
   }
```
Comment (nwingt) [inline pages/in/tokensale/index.vue]:
```css
 padding-right: 0;
 padding-left: 0;
 
 > div {
   margin-right: -8px;
   margin-left: -8px;
 }
```
The above CSS is for achieving overlapping appearance of the progress bar only
I think `tokensale-presale-wrapper` class does not need this in order to preserve default padding

**Comment by williamchong007** [comment]
@nwingt updated

**Pull request merged**